src/librustc_mir/interpret/place.rs

```diff
@@ -926,6 +926,7 @@ where
             // most likey we *are* running `typeck` right now. Investigate whether we can bail out
             // on `typeck_tables().has_errors` at all const eval entry points.
             debug!("Size mismatch when transmuting!\nsrc: {:#?}\ndest: {:#?}", src, dest);
+            self.tcx.sess.delay_span_bug(self.tcx.span, "size-changing transmute");
```

Comment:
Maybe mention how transmute checking should've failed, in the message.
(Because that's the reason this should never actually become an ICE in practice)

Reply:
Also, when I said "better safe than sorry" I meant the current approach, not `delay_span_bug`, but using both is doubly safe :P.

Reply:
Using just `delay_span_bug` is not an option, so I figured the safe variant is to use both. ;)

Reply:
@eddyb updated message